uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementLog.php

```diff
@@ -55,9 +50,7 @@ public static function validateParams($params) {
    */
   private static function validateMandatory($params) {
     $mandatoryFields = [
-      'entitlement_id',
```

Comment:
I'm not sure about making these fields optional. I'm afraid it might lead to inconsistencies in the future if someone forgets to set their values. Also, the FE would also need to remember that we might have a null date (the date is displayed in the changelog). What about defaulting these to the current logged in contact and current date for Leave Period Entitlements that don't have this information?

Reply:
Okay. Thought about this too but was thinking it might be false information because it might not be the current user who made the modification earlier and also the date would be wrong too. 
I can make the change though.